src/fundamental/macro-fundamental.h

```diff
@@ -465,3 +465,10 @@ assert_cc(sizeof(dummy_t) == 0);
 
 assert_cc(STRLEN(__FILE__) > STRLEN(RELATIVE_SOURCE_PATH) + 1);
 #define PROJECT_FILE (&__FILE__[STRLEN(RELATIVE_SOURCE_PATH) + 1])
+
+/* In GCC 14 (C23) we can force enums to have the right types, and not solely rely on language extensions anymore */
+#if ((__GNUC__ >= 14) || (__STDC_VERSION__ >= 202311L))
```

Comment:
Parentheses are not needed here?

Reply:
I copied this from _sd-common.h. I'd prefer to fix these both in a separate commit later